.github/workflows/build.yml

```diff
@@ -12,6 +12,14 @@ env:
   CARGO_TERM_COLOR: always
 
 jobs:
+  QuickFormat:
+    runs-on: ubuntu-latest
+    steps:
+      - name: git-checkout-action
+        uses: actions/checkout@v3
+      - name: dprint-check-action
+        uses: dprint/check@v2.1
+
```

Comment:
I've seen recommendations that tools like dprint are fine, but using them in CI is not as fine, b/c the output of these tools vary widely, even for a given version, on different distros.

how about you do the dprint, and commit what it changes, and it can even be a top level Makefile target or other tool's target, but it's not a check in CI?

Reply:
> ...it can even be a top level Makefile target or other tool's target

I agree we should do this. It was easier to initially get this up and running in CI than digging into the `make + xtask bailiwick`.

A task for myself:

- [ ] Have a top level make task for  the codebase, for developer ease of use. Ideally, the `fmt`ers should be run as a pre-push hook.

> I've seen recommendations that tools like dprint are fine, but using them in CI is not as fine, b/c the output of these tools vary widely, even for a given version, on different distros.

In recent history I'd agree with you on this point. However, in this case we have a stable `fmt`ing toolchain. 

* `cargo install dprint`. `dprint` just has the search mechanics and CLI frontend.
* dprint plugins refs are platfrom independent wasm payloads and are version locked in git. They are also rust programs.
```json
  "plugins": [
    "https://plugins.dprint.dev/json-0.16.0.wasm",
    "https://plugins.dprint.dev/markdown-0.14.1.wasm",
    "https://plugins.dprint.dev/toml-0.5.4.wasm"
  ]
```
* Historically, I agree with you, does this information sway you one way or another?

> how about you do the dprint, and commit what it changes
> (...) but it's not a check in CI?

* IMHO it's easier for everyone for us to be opinionated about the "`fmt`-ing" we accept. Then we don't have 100s or 1000s of whitespace changes based on whomever edited the files last. This makes life (rebase/history/review/merges) easier because change sets have exactly one representation. An exception is for any generated artifacts (lock files for example). 


**...As a middle ground, we can configure the "`fmt`-ing" CI task to be advisory, like a compiler warning.**

Reply:
...I was inspired to do this based on:

https://github.com/oreboot/oreboot/blob/57f2d72e1ff67c54ffd10fc744d69fbd5d21386c/README.md?plain=1#L216-L217